databricks_cli/tunnel/api.py

```diff
@@ -0,0 +1,236 @@
+        self._default_remote_ssh_dir = "~/.ssh"
+        self.cluster_id = None
+        self.context_id = None
+        self.org_id = None
+        self.tunnel_config = None
+        self.debug = debug
+
+    def _init_params(self):
+        self.cluster_id = None
+        self.context_id = None
+        self.org_id = None
+        self.tunnel_config = None
+
+    def check_cluster(self):
+        # check if cluster exist and running
+        cluster_info = self.cluster_client.get_cluster(self.cluster_id)
+        if cluster_info["state"] not in ["RUNNING", "RESIZING"]:
+            raise RuntimeError(f'Cluster with ID: {self.cluster_id} is not RUNNING.')
+        return True
+
+    def get_org_id(self):
+        resp = self.cluster_client.client.client.perform_query("GET", "/clusters/get",
+                                                               data={"cluster_id": self.cluster_id},
+                                                               return_raw_response=True)
+        # TODO(tunneling-cli): this assumes a multitenant workspace, we will need to confirm if
+        #  this exists for st workspace
+        return resp.headers["x-databricks-org-id"]
+
+    def create_context(self):
+        response = self.exec_ctx_client.create_context(language="python",
+                                                       cluster_id=self.cluster_id)
+        return response["id"]
+
+    def destroy_context(self):
+        if self.context_id and self.cluster_id:
+            self.exec_ctx_client.destroy_context(cluster_id=self.cluster_id,
+                                                 context_id=self.context_id)
+
+    def send_public_key_to_driver(self, public_key):
+        print("Sending public key to the driver...")
+        public_key_str = public_key.decode("utf-8")
+        install_keys_on_driver_cmd = f"""
+from pathlib import Path
+
+Path("{self.default_remote_ssh_dir}").expanduser().mkdir(exist_ok=True)
+content = "{public_key_str}"
+with Path("{self.default_remote_ssh_dir}/authorized_keys").expanduser().open("a+") as fp:
+    for line in fp:
+        if content in line:
+            break
+    else:
+        fp.write(content)
+        fp.write("\\n")""".strip()
+        resp = self.command_client. \
+            execute_command_until_terminated(language="python",
+                                             cluster_id=self.cluster_id,
+                                             context_id=self.context_id,
+                                             command=install_keys_on_driver_cmd)
+        if resp["status"] != "Finished":
+            err_data = resp.get("results", {}).get("data", "")
+            raise RuntimeError(f"Sending public key to driver was not successful: {err_data}")
+        if resp["results"]["resultType"] == "error":
+            err_data = resp.get("results", {}).get("cause", "")
+            raise RuntimeError(f"Sending public key to driver was not successful: {err_data})")
```

Comment:
nit: extra parentheses
```suggestion
            raise RuntimeError(f"Sending public key to driver was not successful: {err_data}")
```